utils_cv/tracking/plot.py

```diff
@@ -0,0 +1,156 @@
+from collections import OrderedDict
+import os.path as osp
+from typing import Dict, List, Tuple
+import cv2
+import numpy as np
+import pandas as pd
+
+from .bbox import TrackingBbox
+
+
+def plot_results(
+    results: Dict[int, List[TrackingBbox]],
+    video_path: str,
+    frame_rate: int = 30,
+) -> str:
+    """ 
+    Plot the predicted tracks on the input video. Write the output to track_{input_name}.mp4.
+    
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        video_path: path to the input video
+        frame_rate: frame rate
+
+        Returns (str): 
+            path at which the txt file containing the bboxes and ids in MOT format has been saved.    
+    """
+    # convert results to dataframe in MOT challenge format
+    preds = OrderedDict(sorted(results.items()))
+    bboxes = [
+        [
+            bb.frame_id,
+            bb.track_id,
+            bb.left,
+            bb.top,
+            bb.right - bb.left,
+            bb.bottom - bb.top,
+        ]
+        for _, v in preds.items()
+        for bb in v
+    ]
+
+    df = pd.DataFrame(
+        bboxes, columns=["frame", "id", "left", "top", "width", "height",],
+    )
+
+    data_dir, video_name = osp.split(video_path)
+    output_path = osp.join(data_dir, f"track_{video_name}")
+    _write_video(df, video_path, output_path, frame_rate)
+    print(f"Output saved to {output_path}.")
+
+    return output_path
+
+
+def _write_video(
+    df: pd.DataFrame, input_video: str, output_video: str, frame_rate: int
```

Comment:
Was this function/code copied from somewhere? If not, do we need to use pandas here or could this simply use whatever predict outputs (guess results: Dict[int, List[TrackingBbox]])

Reply:
It's from [https://github.com/microsoft/computervision-recipes/pull/546](https://github.com/microsoft/computervision-recipes/pull/546)

Reply:
I can refactor tho

Reply:
Yeah, please remove all dependencies on Pandas so that it can be used easily and directly with output from the model